### PR TITLE
feat(popup/Settings)!: split settings into tabbed interface

### PR DESCRIPTION
**Pull request by sidvishnoi**
## Context

- Closes https://github.com/interledger/web-monetization-extension/issues/568

### CODE REVIEW

**Comment by sidvishnoi** [comment]
## Screenshots

<details>
<summary>Wallet screen</summary>

</details>

<details>
<summary>Wallet screen, advanced expanded</summary>

</details>

<details>
<summary>Budget screen</summary>

</details>

<details>
<summary>Rate of pay screen</summary>

</details>

<details>
<summary>Rate of pay screen, WM disabled</summary>

</details>

**Comment by sidvishnoi** [comment]
> For the rate of pay, I thought we were getting rid of the slider?

Eventually yes, but not in this PR. It'll get too big, and that input isn't as simple if done right.

> What happens when the user changes the rate of pay or budget amount?

Rate of pay gets handled immediately, like before.
Budget can't be changed at the moment. It's a big change. Those fields are disabled/readonly.

**Comment by RabebOthmani** [comment]
Ok for rate of pay slider. 
As for the budget, we can't just say something is not going to be done because it's big change. The requirement was never for it to be read-only. 
In the last call, we agreed that they can change the budget but it will completely override the previous grant and starts immediately. @raducristianpopa keep me honest here

**Comment by sidvishnoi** [comment]
We'd need UI to support edits first then. We need submit buttons somewhere, and a messaging that they'll need to connect wallet again.

**Comment by RabebOthmani** [comment]
Why do they need to connect to the wallet again?

**Comment by RabebOthmani** [comment]
Does that mean that switching between monthly or not will cause the same issues

**Comment by sidvishnoi** [comment]
When they change budget (any part of it - amount or recurring), we need to create a new interactive grant (i.e. ask user for permission), which happens in a new tab. So, it's the same connect wallet process again, but only without the need to add key again.

**Comment by RabebOthmani** [comment]
You mean the part on the browser to accept the grant? or the connect screen on the extension? 
That also mean that the monthly toggle is the wrong control. A toggle is to make changes not just for visual effects

**Comment by sidvishnoi** [comment]
> You mean the part on the browser to accept the grant?

Yes.

>  That also mean that the monthly toggle is the wrong control. A toggle is to make changes not just for visual effects

Yes, wrong control if it's not meant to be editable. [Point #3 in Slack](https://interledgerfoundation.slack.com/archives/C06CVJRT5N3/p1728391732857209?thread_ts=1728391192.246379&cid=C06CVJRT5N3)

**Comment by RabebOthmani** [comment]
Ok. Settings should be adjustable. A user should have the ability to change the budget allocated to use with the extension and whether or not they want to renew it monthly. 
They will have to accept the grant online. That is absolutely fine. 
We will add a submit changes button that gets enabled if the amount or renew values change. Similar to connect, when they click , they get redirected to the browser. 

Please do let me know if anything is not clear or if there are any technical limitations before deciding on requirements on your own. 
budget

**Comment by sidvishnoi** [comment]
Will do budget-edit support as a follow-up PR. Keeping this PR limited to tab-UI thing only.

File: src/popup/components/Settings/WalletInformation.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  } = useForm();
+
+  return (
+    <div className="flex h-full flex-col gap-8">
+      {/* TODO: Improve error handling */}
+      <form
+        className="space-y-4"
+        onSubmit={handleSubmit(async () => {
+          await message.send('DISCONNECT_WALLET');
+          window.location.reload();
```
Comment (raducristianpopa) [inline src/popup/components/Settings/WalletInformation.tsx:33]: We should check the response here. But I guess it's going to be part of the above TODO item, right?

**Review by raducristianpopa — APPROVED** [review]

**Pull request merged**